lib/runtime/wasmer/imports.go

```diff
@@ -2230,351 +2230,103 @@ func toWasmMemoryFixedSizeOptional(context wasm.InstanceContext, data []byte) (i
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_exists_version_1", ext_storage_exists_version_1, C.ext_storage_exists_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_get_version_1", ext_storage_get_version_1, C.ext_storage_get_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_next_key_version_1", ext_storage_next_key_version_1, C.ext_storage_next_key_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_read_version_1", ext_storage_read_version_1, C.ext_storage_read_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_rollback_transaction_version_1", ext_storage_rollback_transaction_version_1, C.ext_storage_rollback_transaction_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_root_version_1", ext_storage_root_version_1, C.ext_storage_root_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_root_version_2", ext_storage_root_version_2, C.ext_storage_root_version_2)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_set_version_1", ext_storage_set_version_1, C.ext_storage_set_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_storage_start_transaction_version_1", ext_storage_start_transaction_version_1, C.ext_storage_start_transaction_version_1)
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = imports.Append("ext_trie_blake2_256_ordered_root_version_1", ext_trie_blake2_256_ordered_root_version_1, C.ext_trie_blake2_256_ordered_root_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_trie_blake2_256_ordered_root_version_2", ext_trie_blake2_256_ordered_root_version_2, C.ext_trie_blake2_256_ordered_root_version_2)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_trie_blake2_256_root_version_1", ext_trie_blake2_256_root_version_1, C.ext_trie_blake2_256_root_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_trie_blake2_256_verify_proof_version_1", ext_trie_blake2_256_verify_proof_version_1, C.ext_trie_blake2_256_verify_proof_version_1)
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = imports.Append("ext_transaction_index_index_version_1", ext_transaction_index_index_version_1, C.ext_transaction_index_index_version_1)
-	if err != nil {
-		return nil, err
-	}
-	_, err = imports.Append("ext_transaction_index_renew_version_1", ext_transaction_index_renew_version_1, C.ext_transaction_index_renew_version_1)
-	if err != nil {
-		return nil, err
+// ImportsNodeRuntime returns the WASM imports for the node runtime.
+func ImportsNodeRuntime() (imports *wasm.Imports, err error) {
+	imports = wasm.NewImports()
+
+	for _, toRegister := range []struct {
```

Comment:
Would be more clear to have this as a fixed-sized array outside of the function scope IMO

```go
var importedFuncs = [...]struct{
  import_name string
  implementation interface{}
  cgoPointer     unsafe.Pointer
}{
    ...
}

for _, toRegister := importedFuncs {
    _, err = imports.AppendFunction(toRegister.importName, toRegister.implementation, toRegister.cgoPointer)
    if err != nil {
        return nil, fmt.Errorf("importing function: %w", err)
    }
}
```

Reply:
Definitely taking the fixed size array (god damn second time I don't use it, thanks for the reminder!).
Although I don't really see the point to have it at global scope 🤔 I don't think it would change performance wise, and it's only used in that function. It would also pollute more the (unexported) package Go API.